### PR TITLE
[RFC] Move stricmp to vim module and document it

### PR DESCRIPTION
**Pull request by ZyX-I**
Previous location is a leftover of the translator PR which did not intend to run lua plugins in the VimL VM. PR also adds ability to work with NUL strings which was not needed for the translator, but is needed if the goal is to provide case-insensitive string comparison for lua plugins.

### CODE REVIEW

File: src/nvim/lua/executor.c [hunk 1]
```diff
@@ -124,9 +124,41 @@ static void nlua_error(lua_State *const lstate, const char *const msg)
...
+      // Compare "a\0" greater then "a".
+      if ((nul1 == NULL) != (nul2 == NULL)) {
+        ret = ((nul1 != NULL) - (nul2 != NULL));
+        break;
+      }
+      if (nul1 != NULL) {
+        assert(nul2 != NULL);
+        // Due to lowercase letter having possibly different byte length then
+        // uppercase letter can’t shift both strings by the same amount of
+        // bytes.
```
Comment (oni-link) [inline src/nvim/lua/executor.c]: Different byte length means multi-byte characters are used, but can `STRICMP()` handle multi-byte strings?

Comment (ZyX-I) [inline src/nvim/lua/executor.c]: `STRICMP` uses libc strcasecmp by default, this one can if present. Built-in replacement for some reason uses 8-bit libc `tolower` and not Neovim own utf-aware equivalents and thus can’t.

File: runtime/doc/if_lua.txt [hunk 2]
```diff
@@ -246,6 +246,11 @@ Lua interfaces Vim through the "vim" module. Currently it only has `api`
 submodule which is a table with all API functions. Descriptions of these 
 functions may be found in |api.txt|.
```
Comment (justinmk) [inline runtime/doc/if_lua.txt]: "Currently it only has `api` submodule" is  contradicted by `vim.stricmp` now, right?

**Comment by ZyX-I** [comment]
Clarified documentation. CI build skipped because builds for previous commits already succeeded and I only have modified `if_lua.txt`.

File: runtime/doc/if_lua.txt [hunk 3]
```diff
@@ -243,8 +243,49 @@ position are restricted when the command is executed in the |sandbox|.
...
+	Value index for tables representing |Float|s.  A table representing 
+	floating-point value 1.0 looks like this: >
+		{
+			[vim.type_idx] = vim.types.float,
+			[vim.val_idx] = 1.0,
+		}
+<	See also |lua-vim.type_idx| and |lua-special-tbl|.
+
+vim.types						*lua-vim.types*
+	Table with possible values for |lua-vim.type_idx|.  Contains two set
```
Comment (brcolow) [inline runtime/doc/if_lua.txt]: set => sets

**Pull request merged**